test/TCPConnectionTests.cc

```diff
@@ -0,0 +1,254 @@
+/**
+ * @copyright Copyright (c) 2021 ApertureData Inc.
+ */
+
+#include <string>
+#include <thread>
+
+#include "gtest/gtest.h"
+
+#include "Barrier.h"
+#include "ConnClient.h"
+#include "ConnServer.h"
+#include "ExceptionComm.h"
+#include "TCPConnection.h"
+
+#define SERVER_PORT_INTERCHANGE 43444
+#define SERVER_PORT_MULTIPLE    43444
+#define NUMBER_OF_MESSAGES 20
+
+typedef std::basic_string<uint8_t> BytesBuffer;
+
+// Ping-pong messages between server and client
+TEST(TCPConnectionTests, SyncMessages)
+{
+    std::string client_to_server("testing this awesome comm library with some random data");
+    std::string server_to_client("this awesome library seems to work :)");
+
+    Barrier barrier(2);
+
+    std::thread server_thread([&]()
+    {
+        comm::ConnServer server(SERVER_PORT_INTERCHANGE);
+
+        barrier.wait();
+
+        auto server_conn = server.accept();
+
+        for (int i = 0; i < NUMBER_OF_MESSAGES; ++i) {
+            //Recieve something
+            BytesBuffer message_received = server_conn->recv_message();
+            std::string recv_message(reinterpret_cast<char*>(const_cast<uint8_t*>(message_received.data())));
+            ASSERT_EQ(0, recv_message.compare(client_to_server));
```

Comment:
I think the following saves a string copy (same applies throughout this file):
```suggestion
            ASSERT_STREQ( client_to_server.c_str(), reinterpret_cast< const char* >( recv_message.c_str() ) ); 
```

Reply:
mmm the suggested change does not work, I think because of the std::basic_string<uint8_t> and the c_str() method, not sure the casting is correct. We will leave it as is for now.